src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAmendApplicationType.java

```diff
@@ -25,45 +40,85 @@
 import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
 
 @Component
+@RequiredArgsConstructor
 @Slf4j
 public class CaseworkerAmendApplicationType implements CCDConfig<CaseData, State, UserRole> {
 
     public static final String CASEWORKER_AMEND_APPLICATION_TYPE = "caseworker-amend-application-type";
+    private static final Set<DissolveDivorce> DISSOLVE_DIVORCE_SET = Set.of(DissolveDivorce.DISSOLVE_DIVORCE);
+    private static final Set<EndCivilPartnership> END_CIVIL_PARTNERSHIP_SET = Set.of(EndCivilPartnership.END_CIVIL_PARTNERSHIP);
+    private static final Set<JudicialSeparation> JUDICIAL_SEPARATION_SET = Set.of(JudicialSeparation.JUDICIAL_SEPARATION);
+    private static final Set<Separation> SEPARATION_SET = Set.of(Separation.SEPARATION);
+    private final IdamService idamService;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final CcdUpdateService ccdUpdateService;
 
     @Override
     public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
-        configBuilder
+        new PageBuilder(configBuilder
             .event(CASEWORKER_AMEND_APPLICATION_TYPE)
             .forStates(STATES_NOT_WITHDRAWN_OR_REJECTED)
             .name("Amend application type")
             .description("Amend application type")
             .grant(CREATE_READ_UPDATE, SUPER_USER)
             .grantHistoryOnly(LEGAL_ADVISOR, SOLICITOR, CASE_WORKER)
-            .aboutToSubmitCallback(this::aboutToSubmit);
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted))
+            .page("Amend application type")
+            .label("prayerChangeLabel","Completing this event will change the prayer on the case. "
+                + "Please make sure that you have confirmed this with the Applicant before proceeding.");
     }
 
     public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
                                                                        final CaseDetails<CaseData, State> beforeDetails) {
 
         log.info("Caseworker Amend Application Type about to submit callback invoked for case id: {}", details.getId());
         CaseData caseData = details.getData();
+        ApplicantPrayer app1Prayer = caseData.getApplicant1().getApplicantPrayer();
+        ApplicantPrayer app2Prayer = caseData.getApplicant2().getApplicantPrayer();
 
         if (isNull(caseData.getDivorceOrDissolution())) {
             return AboutToStartOrSubmitResponse.<CaseData, State>builder()
                 .data(caseData)
                 .errors(Collections.singletonList("divorceOrDissolution is null, cannot continue submitting event"))
                 .build();
-        } else if (caseData.isJudicialSeparationCase()) {
+        }
+
+        if (caseData.isJudicialSeparationCase()) {
             if (caseData.isDivorce()) {
```

Comment:
think this is getting bit nested now, maybe have private methods for updating the differenct prayers